sklego/meta/zero_inflated_regressor.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any
+
+import numpy as np
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.dummy import DummyClassifier
+from sklearn.linear_model import LinearRegression
+from sklearn.utils.validation import check_is_fitted, check_X_y, check_array
+
+
+class ZeroInflatedRegressor(BaseEstimator, RegressorMixin):
+    """
+    A meta regressor for zero-inflated datasets, i.e. the targets contain a lot of zeroes.
+
+    `ZeroInflatedRegressor` consists of a classifier and a regressor.
+
+        - The classifier's task is to find of if the target is zero or not.
+        - The regressor's task is to output a (usually positive) prediction whenever the classifier indicates that the there should be a non-zero prediction.
+
+    The regressor is only trained on examples where the target is non-zero, which makes it easier for it to focus.
+
+    At prediction time, the classifier is first asked if the output should be zero. If yes, output zero.
+    Otherwise, ask the regressor for its prediction and output it.
+
+    Parameters
+    ----------
+    classifier : Any, scikit-learn classifier, default=`DummyClassifier(strategy="constant", constant=1)`
+        A classifier that answers the question "Should the output be zero?".
+
+    regressor : Any, scikit-learn regressor, default=`LinearRegression()`
```

Comment:
In general, I might prefer `Ridge` over `LinearRegression` but I'm wondering if it's better to not use a default parameter here. It feels like the responsibility of the end-users to come up with something meaningful here that fits their task. Any opportunity to nudge a user to actually think about their problem should be cherished.  😄

Reply:
Let's remove it then together with the classifier.